src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs

```diff
@@ -35,12 +35,31 @@ internal static class WebResponseHelper
             return headers;
         }
 
-        internal static string GetOutFilePath(HttpResponseMessage response, string _qualifiedOutFile)
-        {
-            // Get file name from last segment of Uri
-            string lastUriSegment = System.Net.WebUtility.UrlDecode(response.RequestMessage.RequestUri.Segments[^1]);
+        internal static string GetOutFilePath(HttpResponseMessage response, string qualifiedOutFile)
+        {            
+            if (!Directory.Exists(qualifiedOutFile))
+            {
+                return qualifiedOutFile;
+            }
+
+            string contentDisposition = response.Content.Headers.ContentDisposition?.FileNameStar ?? response.Content.Headers.ContentDisposition?.FileName;
```

Comment:
Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition#directives the value may have path information and should be stripped.  It may or may not be enclosed in quotes.

Reply:
The guideline says that "filename" is usually default name for "Save As". This force we think that we go in wrong direction trying to make the cmdlet too smart. We don't really know what the user wants in a particular scenario. Therefore, I propose follow @mklement0 's proposal  and introduce a new parameter AutoFileName as enum:
- Default. OutFile works as usually and requires exact path and file name.
- URL.  OutFile is a directory and file name is generated based on URL. Throw like curl if no last segment.
- Form. OutFile is a directory and file name is generated based on Content-Disposition. Throw if no Content-Disposition or filename in Content-Disposition.
This way we allow the user to do exactly what he needs to do.

Reply:
I'd prefer if we left the decision to the cmdlet, if the user wants to choose a name for the file he just has to write it.

Reply:
@SteveL-MSFT what should be the best course of action?

Reply:
@SteveL-MSFT any news on AutoFileName?

Reply:
@CarloToso sorry for the delay, I'll queue this up for discussion in Cmdlets WG